### PR TITLE
DATA-841 Add rdk representation of worldstate along with conversion methods

### PR DESCRIPTION
**Pull request by raybjork**
This PR removes `commonpb.WorldState` from most of the codebase and replaces it with `referenceframe.WorldState` a representation that is more easily usable with existing RDK functions.

### CODE REVIEW

File: referenceframe/worldstate.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	"github.com/pkg/errors"
+	commonpb "go.viam.com/api/common/v1"
+
+	spatial "go.viam.com/rdk/spatialmath"
+)
+
+// WorldState is a struct to store the data representation of the robot's environment.
+type WorldState struct {
+	Obstacles, InteractionSpaces []*GeometriesInFrame
+	Transforms                   []*commonpb.Transform
```
[the remark this answers is not in the document]
Comment (biotinker) [reply referenceframe/worldstate.go:15]: PoseInFrame, not static frame, sorry

File: referenceframe/transformable.go [hunk 2]
```diff
@@ -55,15 +64,67 @@ func PoseInFrameToProtobuf(framedPose *PoseInFrame) *commonpb.PoseInFrame {
...
+		return nil, ErrEmptyStringFrameName
+	}
+	poseMsg := poseInObserverFrame.GetPose()
+	pose := spatialmath.NewPoseFromProtobuf(poseMsg)
+	return NewNamedPoseInFrame(parentFrame, pose, frameName), nil
+}
+
+// PoseInFrameSliceToTransformProtobuf converts a slice of PoseInFrame structs to a slice of Transform protobuf messages.
+func PoseInFrameSliceToTransformProtobuf(poseSlice []*PoseInFrame) ([]*commonpb.Transform, error) {
+	protoTransforms := make([]*commonpb.Transform, 0)
```
Comment (biotinker) [inline referenceframe/transformable.go:104]: nit: Should be `make([]*commonpb.Transform, 0, len(poseSlice))`

File: referenceframe/transformable.go [hunk 3]
```diff
@@ -55,15 +64,67 @@ func PoseInFrameToProtobuf(framedPose *PoseInFrame) *commonpb.PoseInFrame {
...
+			return nil, err
+		}
+		protoTransforms = append(protoTransforms, protoTf)
+	}
+	return protoTransforms, nil
+}
+
+// PoseInFrameSliceFromTransformProtobuf converts a slice of Transform protobuf messages to a slice of PoseInFrame structs.
+func PoseInFrameSliceFromTransformProtobuf(proto []*commonpb.Transform) ([]*PoseInFrame, error) {
+	transforms := make([]*PoseInFrame, 0)
```
Comment (biotinker) [inline referenceframe/transformable.go:117]: nit: Should be `make([]*PoseInFrame, 0, len(proto))`

**Review by biotinker — APPROVED** [review]
Two small nits, then LGTM

File: referenceframe/worldstate.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (raybjork) [reply referenceframe/worldstate.go]: Done

**Review by gvaradarajan — CHANGES_REQUESTED** [review]

File: robot/framesystem/service_test.go [hunk 4]
```diff
@@ -48,44 +46,15 @@ func TestFrameSystemFromConfig(t *testing.T) {
...
-			},
-		},
-		{
-			ReferenceFrame: "frame3",
-			PoseInObserverFrame: &commonpb.PoseInFrame{
-				ReferenceFrame: "world",
-				Pose:           spatialmath.PoseToProtobuf(testPose),
-			},
-		},
+	transforms := []*referenceframe.PoseInFrame{
```
[the remark this answers is not in the document]
Comment (raybjork) [reply robot/framesystem/service_test.go:49]: Thanks!

File: referenceframe/transformable.go [hunk 5]
```diff
@@ -55,15 +69,67 @@ func PoseInFrameToProtobuf(framedPose *PoseInFrame) *commonpb.PoseInFrame {
...
+		if err != nil {
+			return nil, err
+		}
+		protoTransforms = append(protoTransforms, protoTf)
+	}
+	return protoTransforms, nil
+}
+
+// PoseInFrameSliceFromTransformProtobuf converts a slice of Transform protobuf messages to a slice of PoseInFrame structs.
+func PoseInFrameSliceFromTransformProtobuf(protoSlice []*commonpb.Transform) ([]*PoseInFrame, error) {
```
[the remark this answers is not in the document]
Comment (raybjork) [reply referenceframe/transformable.go:121]: changed

File: referenceframe/transformable.go [hunk 6]
```diff
@@ -55,15 +69,67 @@ func PoseInFrameToProtobuf(framedPose *PoseInFrame) *commonpb.PoseInFrame {
...
+	pose := spatialmath.NewPoseFromProtobuf(poseMsg)
+	return NewNamedPoseInFrame(parentFrame, pose, frameName), nil
+}
+
+// PoseInFrameSliceToTransformProtobuf converts a slice of PoseInFrame structs to a slice of Transform protobuf messages.
+func PoseInFrameSliceToTransformProtobuf(poseSlice []*PoseInFrame) ([]*commonpb.Transform, error) {
+	protoTransforms := make([]*commonpb.Transform, 0, len(poseSlice))
+	for _, transform := range poseSlice {
+		protoTf, err := PoseInFrameToTransformProtobuf(transform)
+		if err != nil {
```
[the remark this answers is not in the document]
Comment (raybjork) [reply referenceframe/transformable.go:112]: done

**Review by gvaradarajan — APPROVED** [review]

**Pull request merged**